connection_util.go

```diff
@@ -194,10 +194,13 @@ func getResultType(ctx context.Context) resultType {
 
 // isDml returns true if the statement type code is in the range of DML.
 func isDml(v int64) bool {
-	return statementTypeIDDml <= v && v <= statementTypeIDMultiTableInsert
+	return (statementTypeIDDml <= v && v <= statementTypeIDMultiTableInsert) || v == statementTypeIDSelect
```

Comment:
Can you explain the reason of this change? I believe that SELECT is not a DML query.

Reply:
It seems that you are right as in Snowflake it is not considered as one as per docs, however I have found a multiple mentions in the internet that it is considered a limited DML, i.e.:
- https://docs.oracle.com/cd/B13789_01/server.101/b10759/statements_1001.htm#:~:text=The%20SELECT%20statement%20is%20a,the%20results%20of%20the%20query.
- https://en.wikipedia.org/wiki/Data_manipulation_language